common/archiver/gcloud/historyArchiver_test.go

```diff
@@ -0,0 +1,470 @@
+	mockCtrl := gomock.NewController(h.T())
+	URI, err := archiver.NewURI("gs://my-bucket-cad/cadence_archival/development")
+	storageWrapper := &mocks.Client{}
+	storageWrapper.On("Exist", ctx, URI, "").Return(true, nil).Times(1)
+	storageWrapper.On("Query", ctx, URI, mock.Anything).Return([]string{"905702227796330300141628222723188294514017512010591354159_-24_0.history", "905702227796330300141628222723188294514017512010591354159_-25_0.history"}, nil).Times(1)
+	storageWrapper.On("Get", ctx, URI, "71817125141568232911739672280485489488911532452831150339470_-24_0.history").Return([]byte(exampleHistoryRecord), nil)
+	historyIterator := archiver.NewMockHistoryIterator(mockCtrl)
+	historyArchiver := newHistoryArchiver(h.container, historyIterator, storageWrapper)
+	request := &archiver.GetHistoryRequest{
+		DomainID:   testDomainID,
+		WorkflowID: testWorkflowID,
+		RunID:      testRunID,
+		PageSize:   testPageSize,
+	}
+
+	h.NoError(err)
+	response, err := historyArchiver.Get(ctx, URI, request)
+	h.NoError(err)
+	h.Nil(response.NextPageToken)
+}
+
+func (h *historyArchiverSuite) TestGet_Success_UseProvidedVersion() {
+
+	ctx := context.Background()
+	mockCtrl := gomock.NewController(h.T())
+	URI, err := archiver.NewURI("gs://my-bucket-cad/cadence_archival/development")
+	storageWrapper := &mocks.Client{}
+	storageWrapper.On("Exist", ctx, URI, "").Return(true, nil).Times(1)
+	storageWrapper.On("Query", ctx, URI, "71817125141568232911739672280485489488911532452831150339470").Return([]string{"905702227796330300141628222723188294514017512010591354159_-24_0.history", "905702227796330300141628222723188294514017512010591354159_-25_0.history"}, nil).Times(1)
+	storageWrapper.On("Get", ctx, URI, "71817125141568232911739672280485489488911532452831150339470_1_0.history").Return([]byte(exampleHistoryRecord), nil)
+	historyIterator := archiver.NewMockHistoryIterator(mockCtrl)
+	historyArchiver := newHistoryArchiver(h.container, historyIterator, storageWrapper)
+	request := &archiver.GetHistoryRequest{
+		DomainID:             testDomainID,
+		WorkflowID:           testWorkflowID,
+		RunID:                testRunID,
+		PageSize:             testPageSize,
+		CloseFailoverVersion: common.Int64Ptr(1),
+	}
+
+	h.NoError(err)
+	response, err := historyArchiver.Get(ctx, URI, request)
+	h.NoError(err)
+	h.Nil(response.NextPageToken)
+}
+
+func (h *historyArchiverSuite) SetupTest() {
+	zapLogger := zap.NewNop()
+	h.Assertions = require.New(h.T())
+	h.container = &archiver.HistoryBootstrapContainer{
+		Logger:        loggerimpl.NewLogger(zapLogger),
+		MetricsClient: metrics.NewClient(tally.NoopScope, metrics.History),
+	}
+	h.testArchivalURI, _ = archiver.NewURI("gs://my-bucket-cad/cadence_archival/development")
+}
+
+func TestHistoryArchiverSuite(t *testing.T) {
+	suite.Run(t, new(historyArchiverSuite))
+}
+
+type historyArchiverSuite struct {
+	*require.Assertions
+	suite.Suite
+	container       *archiver.HistoryBootstrapContainer
+	testArchivalURI archiver.URI
+}
```

Comment:
Please move them to the top of the file.

Reply:
done